Add registry settings admin task to access guide

smccarthy-ie opened the pull request:
- Add admin task for registry instance settings to access guide 
- Fix broken link to registry CLI guide in top-level readme
- Add peer review feedback on entire SR access guide for consistency

wtrocki commented:
LGTM

wtrocki commented:
Obviously need to be timed with the release (subject to +3-4 days)

bredamc commented on docs/registry/access-mgmt-registry/README.adoc:
```diff
@@ -100,10 +100,11 @@ An account in {product-long-registry} is either a user account or a service acco
 
-* *Administrator* - Can perform the following in this {registry} instance:
-** View or write user roles
+* *Administrator* - Can perform the following tasks in this {registry} instance:
```
```suggestion
* *Administrator* 
Users with the Administrator role can perform the following tasks in this {registry} instance:
```

I might not have coded it correctly, but basically I'm suggesting using a term definition list, where the term is on a line of its own to make it easier to read.

Also rephrased to use a complete sentence.

smccarthy-ie replied:
Sure, I think you mean a description list? https://docs.asciidoctor.org/asciidoc/latest/lists/description/

So format is `Administrator:: Users with the Administrator role...`

bredamc replied:
Perfect, thanks! It's called a _definition list_ in IBM Style, see the Lists topic.

bredamc commented on docs/registry/access-mgmt-registry/README.adoc:
```diff
@@ -100,10 +100,11 @@ An account in {product-long-registry} is either a user account or a service acco
+** Configure user roles for access
+** Configure {registry} settings
+** Configure global rules for artifact compatibility and validity
 ** Import/export {registry} data
```
```suggestion
** Import or export {registry} data
```
Avoid the use of slashes -- can be ambiguous for translators. See the Slashes topic in IBM Style.

bredamc commented on docs/registry/access-mgmt-registry/README.adoc:
```diff
@@ -100,10 +100,11 @@ An account in {product-long-registry} is either a user account or a service acco
+** Configure {registry} settings
+** Configure global rules for artifact compatibility and validity
 ** Import/export {registry} data
 * *Manager* - Can view or write schema and API artifacts in this {registry} instance
```
```suggestion
* *Manager* 
Users with the Manager role can view or write schema and API artifacts in this {registry} instance.
```

bredamc commented on docs/registry/access-mgmt-registry/README.adoc:
```diff
@@ -100,10 +100,11 @@ An account in {product-long-registry} is either a user account or a service acco
+** Configure global rules for artifact compatibility and validity
 ** Import/export {registry} data
 * *Manager* - Can view or write schema and API artifacts in this {registry} instance
 * *Viewer* - Can view schema and API artifacts in this {registry} instance
```
```suggestion
* *Viewer* 
Users with the Viewer role can view schema and API artifacts in this {registry} instance.
```

bredamc reviewed (commented):
Some minor comments

bredamc commented:
Some additional comments (also sent by email, where the coloured text and strikethrough text makes it easier to read):

1. "Access management in Red Hat OpenShift Service Registry", second para:
    Delete extra words:
    "A service account enables your client application or tool to connect securely to your Service Registry instances."

2. "Access ..." section, Important note at the end of the section:
    "administrator role" should be "Administrator role" (initcap).

3. "Viewing user roles in a Service Registry instance", first sentence:
    Change "that" to "to":
    "You can view the user roles assigned to your Service Registry instances to manage how other user accounts ..."

4. Global, all prereqs:
    Use simpler tense:
    Example: "The Service Registry instance is created and is in **Ready** state."

5. Global suggestion:
    Add the standard prereq for accessing the RHOSR console (with the link), and then simplify the first step in each procedure (cf. the Getting Started)?

6. "Viewing ...", step 2b:
    "To view accounts with a specific role, click **Role**, click **Filter by role** and then select the role ... "

7. "Viewing ...", step 3:
    Rephrase to orient the reader first:
    "When you are finished, click **Clear all filters**."

8. "Assigning...", step 4:
    Append a fullstop at the end of each bullet point.

9. "Editing ...", first para:
    Use consistent text for the intro para in each of the main sections:
    (a) Are the roles "assigned in" or "assigned to" the instances?
    (b) Change "that manage" to "to manage"? ("that" is ambiguous, does it refer to the role or the instance.)

10. "Editing", Step 3a:
    Split into two sentences:
    "...click **Edit**. Select the new user role, for example, click **Viewer** for read-only access to this instance, and click **Save**."

11. "Editing", Step 3b:
    Split into two sentences:
    "... click **Remove**. Click **Remove** again to confirm."

12. "Allowing ...", first para:
    Change "Red Hat web console" to "Hybrid Cloud Console"?

13. "Allowing ...", second para:
    "... users within the group can do the following tasks:"

 14. "Allowing ...", Prereqs:
    The link to the document opens the splash page instead of the specified doc:
    "[User Access Configuration Guide for Role-based Access Control (RBAC)](https://access.redhat.com/documentation/en-us/red_hat_hybrid_cloud_console/)"

15. "Allowing ...", step 1:
    Change "In the upper-right corner ..." to "At the top ...", and append a fullstop.

16. "Allowing ...", step 1:
    I wonder if we should separate the clicking of **Settings** (from the dropdown menu after you click the gear icon) and the clicking of **User Access > Groups** from the left nav pane afterwards? Maybe it's better as is?

17. "Allowing ...", steps 3 & 4:
    Remove "to add the role to the group" from both steps.

18. "Allowing ...", step 4:
    In the second sentence, maybe add "also", if that's a separate thing? Or add the "group" text there:
    "The role is added to the group, and to the list of selected roles ..."


That's it -- as usual, feel free to disregard any that you don't agree with :)

bredamc commented:
Re point 10: Don't forget to include an instruction to click Save.

Re. point 14: In case you thought I had provided the direct link in my original comment -- I had copied the existing link, which points to the splash page. Not sure if you are deliberately linking to the splash page, or if you want to link to the doc directly here:
 https://access.redhat.com/documentation/en-us/red_hat_hybrid_cloud_console/2022/html/user_access_configuration_guide_for_role-based_access_control_rbac/index

smccarthy-ie commented:
Thanks @bredamc. I've addressed your feedback on the initial commit, plus the additional peer review feedback on the entire guide. 

I've implemented pretty much everything in the guide review except #14. I like the way it currently links to the splash page so I can see the other docs in HCC library for additional context. I'm also guessing the original writer may have done it this way to minimize risk of broken links.

bredamc reviewed (approved):
LGTM :)
